src/server/controller/assessment/generateMetaCache/dependencyEvaluator/evalDependencies/member.ts

```diff
@@ -12,35 +13,47 @@ const excludeDependant = (row: Row, tableName: string, variableName: string): bo
 
 export class MemberEvaluator extends ExpressionNodeEvaluator<Context, MemberExpression> {
   evaluate(expressionNode: MemberExpression): string {
-    const { object, property } = expressionNode
-
     const { assessmentMetaCache, row, tableName, type } = this.context
 
-    // @ts-ignore
-    const objectName = object?.object?.name ?? object.name
-    // @ts-ignore
-    const propertyName = object?.property?.name ?? property.name
+    // Example of: <calcualteFn> = [...result]
+    // Case 1: extentOfForest.forestArea (2) [extentOfForest, forestArea]
+    // Case 2: extentOfForest.forestArea['2025'] (3) [extentOfForest, forestArea]
+    // Case 3: fra['2025'].extentOfForest.forestArea (4) [fra, '2025', extentOfForest, forestArea]
+    // Case 4: fra['2025'].extentOfForest.forestArea['2025'] (4) [fra, '2025', extentOfForest, forestArea]
+
+    const {
+      tableName: _tableName,
+      variableName: _variableName,
+      assessmentName,
+      cycleName,
+    } = ExpressionEvaluator.parseExpression(expressionNode)
 
-    if (assessmentMetaCache.variablesByTable[objectName]) {
-      const dependantTable = assessmentMetaCache[type].dependants?.[objectName] ?? {}
-      const dependants = dependantTable[propertyName] ?? []
+    if (assessmentMetaCache.variablesByTable[_tableName]) {
```

Comment:
1. it's a but unclear reading _tableName and tableName. it's difficult to understand what is what.
2. parseExpression sounds like a public method that does something different from what it really does. call this parseMemberVariable and don't expose it from ExpressionEvaluator. is it just an internal method used only here, right?

the result of my 2 comments above could be:

```
const memberVariable  = parseMemberVariable(expressionNode)
```

Reply:
Used in 2 places:
https://github.com/openforis/fra-platform/blob/89786310e0db4026c01525d239ab7f267a2c52bb/src/meta/expressionEvaluator/evaluators/member.ts#L8

https://github.com/openforis/fra-platform/blob/89786310e0db4026c01525d239ab7f267a2c52bb/src/server/controller/assessment/generateMetaCache/dependencyEvaluator/evalDependencies/member.ts#L20-L32

Reply:
I only see one usage, and one import

Reply:
![image](https://github.com/openforis/fra-platform/assets/5508251/02db2d9a-a927-4019-a9fd-b01bd22a5c20)
![image](https://github.com/openforis/fra-platform/assets/5508251/0c9e1488-2ea7-4685-a5a8-92cdcac82c67)

Reply:
should i still export only the function?